aten/src/ATen/cpu/vec256/vec256_bfloat16.h

```diff
@@ -194,25 +194,39 @@ template <> class Vec256<BFloat16> {
 
 };
 
-#define DEFINE_BF16_OP(op, func)                                                            \
-template<>                                                                                  \
-Vec256<BFloat16> inline operator op(const Vec256<BFloat16>& a, const Vec256<BFloat16>& b) { \
-  __m256 a_lo, a_hi;                                                                        \
-  __m256 b_lo, b_hi;                                                                        \
-  cvtbf16_fp32(__m256i(a), a_lo, a_hi);                                                     \
-  cvtbf16_fp32(__m256i(b), b_lo, b_hi);                                                     \
-  auto o1 = func(a_lo, b_lo);                                                               \
-  auto o2 = func(a_hi, b_hi);                                                               \
-  return cvtfp32_bf16(o1, o2);                                                              \
+template<typename Op>
+Vec256<BFloat16> inline bfloat16_binary_op_as_fp32(const Vec256<BFloat16>& a, const Vec256<BFloat16>& b, Op op) {
```

Comment:
@XiaobingSuper Do you think this function can also be used for implementing operators `>`, `<`, `>=`, and `<=`? Now #35117 should be waiting on these operators.